### PR TITLE
Add data about variants in debug mode

### PR DESCRIPTION
**Pull request by alehander92**
Variant debug info

### CODE REVIEW

File: compiler/cgendata.nim [hunk 1]
```diff
@@ -155,6 +155,8 @@ type
     labels*: Natural          # for generating unique module-scope names
     extensionLoaders*: array['0'..'9', Rope] # special procs for the
                                               # OpenGL wrapper
+    variants*: HashSet[string] # debug info for variants
```
Comment (Araq) [inline compiler/cgendata.nim]: This should be a ``HashSet[SigHash]``.

**Comment by Araq** [comment]
I don't understand why this is required, the RTTI already contains all the information you need.

**Comment by alehander42** [comment]
@Araq does it? I wasn't sure, if it does, I'll gladly close it

What I need is to be able to find which is the active "union" in a variant just based on the value of the kind(or other) field: is this easily possible with the current RTTI?

**Comment by Araq** [comment]
Check how ``system.selectBranch`` is used.

**Comment by alehander42** [comment]
hm, actually I am not sure selectBranch directly will help me, because I want to be able to find out the possible mappings only having the type name, but I can find this with NimDT name fields in a way, so I'll close it if this works instead.

On the other hand we need to always generate NimDT/enum NTI in debug mode, I'll open a different PR for that

**Comment by Araq** [comment]
Stalled PR, reopen if it's important.

**Pull request closed without merging**